src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,322 @@
   onDataSaved: () => void;
+  onGoBack: () => void;
   user: Session["user"];
 };
 
-export const EnterInfo = (_props: Props) => {
-  const _l10n = useL10n();
+// TODO: Add more sophisticated for location data
+const getIsValidInfo = (value: string) => value !== "";
+
+export const EnterInfo = (props: Props) => {
+  const hasMounted = useRef(false);
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [location, setLocation] = useState("");
+  const [dateOfBirth, setDateOfBirth] = useState("");
+
+  const [invalidInputs, setInvalidInputs] = useState<Array<string>>([]);
+
+  const explainerDialogState = useOverlayTriggerState({});
+  const explainerDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    explainerDialogState
+  );
+
+  const confirmDialogState = useOverlayTriggerState({});
+  const confirmDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    confirmDialogState
+  );
+
+  const l10n = useL10n();
+
+  const userDetailsData = [
+    {
+      label: l10n.getString("onboarding-enter-details-label-first-name"),
+      key: "firstName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-first-name"
+      ),
+      value: firstName,
+      displayValue: firstName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setFirstName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-last-name"),
+      key: "lastName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-last-name"
+      ),
+      value: lastName,
+      displayValue: lastName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLastName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-location"),
+      key: "location",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-location"
+      ),
+      value: location,
+      displayValue: location,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLocation,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-date-of-birth"),
+      key: "dateOfBirth",
+      type: "date",
+      placeholder: "",
+      value: dateOfBirth,
+      displayValue: new Date(dateOfBirth).toLocaleDateString("en-US"),
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setDateOfBirth,
+    },
+  ];
+
+  const validateEnteredInfo = () => {
+    const invalidInputKeys = userDetailsData
+      .filter(({ value }) => !getIsValidInfo(value))
+      .map(({ key }) => key);
+
+    setInvalidInputs(invalidInputKeys);
+  };
+
+  useEffect(() => {
+    if (!hasMounted.current) {
```

Comment:
This could use a comment to clarify what's going on here :) (And also maybe put the `useRef` on the line above?)

I'm particularly interested in understanding why you wouldn't be able to call the following in `validateEnteredInfo`:

```js
    if (invalidInputKeys.length === 0) {
      confirmDialogState.open();
    }
```

Reply:
Made it too complicated and removed it together with `useEffect` in [1b294a9](https://github.com/mozilla/blurts-server/pull/3208/commits/1b294a9bbe3570e375e9259339b2b1121b036433).